contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -163,18 +168,39 @@ fn apply_to_hub(
 
 fn apply_to_transfer(
     storage: &mut dyn Storage,
+    querier: QuerierWrapper,
     source_chain: ChainNameRaw,
     destination_chain: ChainNameRaw,
     transfer: InterchainTransfer,
 ) -> Result<InterchainTransfer, Error> {
-    interceptors::subtract_supply_amount(storage, &source_chain, &transfer)?;
-    let transfer = interceptors::apply_scaling_factor_to_amount(
-        storage,
-        &source_chain,
-        &destination_chain,
-        transfer,
-    )?;
-    interceptors::add_supply_amount(storage, &destination_chain, &transfer)?;
+    // we only do balance tracking for amplifier chains
+    let client: nexus::Client = client::CosmosClient::new(querier).into();
+    let source_is_amplifier_chain = !client
+        .is_chain_registered(&source_chain.normalize())
+        .change_context(Error::Nexus(source_chain.clone()))?;
+    let destination_is_amplifier_chain = !client
+        .is_chain_registered(&destination_chain.normalize())
+        .change_context(Error::Nexus(destination_chain.clone()))?;
+
+    if source_is_amplifier_chain {
+        interceptors::subtract_supply_amount(storage, &source_chain, &transfer)?;
+    }
```

Comment:
This is very procedural and is exactly what the interceptors PR wanted to avoid. I'd much prefer this function to just look like
```
interceptors::subtract_supply_amount_if_amplifier(storage, querier, &source_chain, &transfer)?;
    let transfer = interceptors::apply_scaling_factor_to_amount_if_amplifier(
        storage,
        querier,
        &source_chain,
        &destination_chain,
        transfer,
    )?;
    interceptors::add_supply_amount_if_amplifier(storage, querier, &destination_chain, &transfer)?;
```

Reply:
I don't think we should pass the querier in, because then we are going to need to repeatedly query whether a chain is registered with core. I can just query first and pass that data in though

Reply:
With the `ChainSpecifier`, it looks even worse now. We repeated loading/querying/saving a lot of stuff. If gas cost is the top priority, we wouldn't be loading source token instance, subtracting, saving, loading destination token instance, adding, saving, loading both token instances again, applying scaling factor.